pkg/cloud/vsphere/provisioner/govmomi/delete.go

```diff
@@ -20,52 +21,68 @@ func (pv *Provisioner) Delete(ctx context.Context, cluster *clusterv1.Cluster, m
 	if cluster == nil {
 		return errors.New(constants.ClusterIsNullErr)
 	}
+	if exists, _ := pv.Exists(ctx, cluster, machine); exists {
+		if err := pv.powerOffAndDestroy(ctx, cluster, machine); err != nil {
+			return err
+		}
+	}
+
+	// Delete the node object bound to the Machine via the Status.NodeRef reference
+	if machine.Status.NodeRef != nil {
+		// Do a force delete (grace period set to 0) as the underlying VM is already deleted
+		err := pv.k8sClient.Core().Nodes().Delete(machine.Status.NodeRef.Name, metav1.NewDeleteOptions(0))
+		if err != nil {
```

Comment:
we could let controller re-try by returning `controllerError.RequeueAfterError`, my thinking is we should try to achieve consistent state, ignore the error (throw warning) means: node deletion may success or fail, it will be two possiable states.